kernel/include/timeout_q.h

```diff
@@ -52,9 +51,9 @@ static inline void z_init_thread_timeout(struct _thread_base *thread_base)
 
 extern void z_thread_timeout(struct _timeout *timeout);
 
-static inline void z_add_thread_timeout(struct k_thread *thread, k_timeout_t ticks)
+static inline k_ticks_t z_add_thread_timeout(struct k_thread *thread, k_timeout_t ticks)
```

Comment:
Since this is now returning a value with meaning, I think that it is worthwhile to note what this return value means (system tick when timeout is expected to expire).